jcl/src/java.base/share/classes/java/lang/ClassLoader.java

```diff
@@ -1407,6 +1411,12 @@ private static boolean registerAsParallelCapable(Class<?> callerCls) {
 
 	Class<?> superCls = callerCls.getSuperclass();
 
+	/*[IF JAVA_SPEC_VERSION >= 19]*/
+	if (!ClassLoader.class.isAssignableFrom(superCls)) {
```

Comment:
Shouldn't this use `callerCls` instead of `superCls`?

Reply:
This only works because `ClassLoader` itself is abstract: that subtlety deserves a comment, IMHO.

Reply:
Created https://github.com/eclipse-openj9/openj9/pull/15312